bundle-plugin/src/main/java/com/yahoo/container/plugin/mojo/GenerateOsgiManifestMojo.java

```diff
@@ -118,8 +118,13 @@ public void execute() throws MojoExecutionException {
                         .map(e -> "(" + e.getPackageNames().toString() + ", " + e.version().orElse("")).collect(Collectors.joining(", ")));
             }
 
-            // TODO: skip if jdisc_core is not in class path?
-            logMissingPackages(publicPackagesFromProvidedJars, projectPackages, compileJarsPackages, includedPackages);
+            if (hasJdiscCoreProvided(artifactSet.getJarArtifactsProvided())) {
+                // If jdisc_core is not provided, log output may contain packages that _are_ available runtime.
+                logMissingPackages(publicPackagesFromProvidedJars, projectPackages, compileJarsPackages, includedPackages);
+            } else {
+                getLog().warn("This project does not have jdisc_core as provided dependency, so the " +
```

Comment:
FYI, I know about many Maven modules in the Vespa repo that is a bundle but does not include jdisc_core.

Reply:
Almost all internal bundles depend on `container-dev` which pulls in `jdisc_core`. AFAIK, only yolean and security-utils do not, although there could be a few more. For bundles in general (and user bundles in particular) they must have jdisc_core to get necessary Import-Packages.